cmd/lifecycle/analyzer.go

```diff
@@ -274,39 +253,61 @@ func (a *analyzeCmd) supportsRunImage() bool {
 	return a.platformAPIVersionGreaterThan06()
 }
 
-func (a *analyzeCmd) validateRunImageInput() error {
-	if !a.supportsRunImage() && a.runImageRef != "" {
-		return errors.New("-run-image is unsupported")
```

Comment:
Did this check disappear on purpose?

Reply:
I realized that this is already covered by `cmd.FlagRunImage(&a.runImageRef)` being scoped to the `if` block in `DefineFlags()` (the error message is `flag provided but not defined: -run-image`). Maybe we could add an acceptance test for it. It's probably also worth adding a comment for `DefineFlags()` because just glancing at the function (without digging into the internals of `FlagSet`) it just looks like it's reading values into memory when it's also defining which flags are valid.

Reply:
Added comments in d1988a83